pipeline-scripts/release.groovy

```diff
@@ -21,24 +22,61 @@ def stageValidation(quay_url, name, advisory) {
         error("Payload ${name} already exists! Cannot continue.")
     }
 
-    // AMH - This may be optional?
-    if (advisory != "") {
-        echo "Verifying advisory exists"
-        res = commonlib.shell(
+    if (!advisory) {
+        echo "Getting current advisory for OCP $version from build data..."
+        res = withEnv(["_VERSION=$version"]) {
+            commonlib.shell(
                 returnAll: true,
-                script: "elliott get ${advisory}"
-        )
+                script: 'elliott --group=openshift-"${_VERSION}" get --json - --use-default-advisory image'
+            )
+        }
+        if(res.returnStatus != 0) {
+            error("🚫 Advisory number for OCP $version couldn't be found from ocp_build_data.")
+        }
+    } else {
+        echo "Verifying advisory ${advisory} exists"
+        res = withEnv(["_VERSION=$version", "_ADVISORY=$advisory"]) {
+            commonlib.shell(
+                returnAll: true,
+                script: 'elliott --group=openshift-"${_VERSION}" get --json - -- "${_ADVISORY}"'
+            )
+        }
 
         if(res.returnStatus != 0){
             error("Advisory ${advisory} does not exist! Cannot continue.")
         }
     }
+
+    def advisoryInfo = readJSON text: res.stdout
+    retval.advisoryInfo = advisoryInfo
+
+    echo "Verifying advisory ${advisoryInfo.id} (https://errata.engineering.redhat.com/advisory/${advisoryInfo.id}) status"
+    if (advisoryInfo.status != 'QE') {
+        error("🚫 Advisory ${advisoryInfo.id} is not in QE state.")
+    }
+    echo "✅ Advisory ${advisoryInfo.id} is in QE state."
+
+    // Extract liveID (if exists) from advisory info
+    // FIXME: Not sure if this is a stable way.
+    // If the advisory has a live ID assigned, let's say RHSA-2019:2594-03,
+    // advisoryInfo.errata_id should be 2547 and advisoryInfo.fulladvisory be RHSA-2019:2594-03
```

Comment:
do you mean `2594` I don't understand where this 2547 from?

Reply:
This is a mistake. I will fix it.

Reply:
Done